lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -134,8 +134,8 @@ sharedToLinearLayoutNoLeadingOffset(ArrayRef<int64_t> shape,
   // Construct bases for the 2 most minor dimensions of the layout.  These are
   // the dims that get swizzled.
   assert(shape.size() >= 2);
-  int colDim = shared.getOrder()[0];
-  int rowDim = shared.getOrder()[1];
+  int colDim = getDefaultOrder(shared)[0];
```

Comment:
For example, it's not clear to me why we do not call `getOrder` here

Reply:
As described in the OP, the idea is that `getDefaultOrder` should (in the future) just called in `LinearLayoutConversions.cpp`. It's basically just used in the definition of the LL.

For shared layouts, this may or may not be necessary, as we don't broadcast them over the size, so perhaps we could simply implement `getOrder` manually and call it a day.

I'm happy to do it that way, so that we just touch distributed for now.